src/types/polymorphic.ts

```
@@ -0,0 +1,38 @@
+export type Merge<M, N> =
+  N extends Record<string, unknown> ? M : Omit<M, keyof N> & N;
+
+type AsProp<T extends ElementType> = {
+  as?: T;
+};
+
+type PropsToOmit<T extends ElementType, P> = keyof (AsProp<T> & P);
+
+export type PolymorphicComponentProps<
+  T extends ElementType,
+  Props = {},
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)